types/react-dom/test-utils/index.d.ts

```diff
@@ -0,0 +1,354 @@
+ * @template T
+ * @template C
+ * @param {ReactInstance} instance
+ * @param {ClassType<any, T, C>} type
+ * @returns {T}
+ */
+export function isCompositeComponentWithType<T extends Component<any, any>, C extends ComponentClass<any>>(
+    instance: ReactInstance, type: ClassType<any, T, C>): boolean;
+
+/**
+ * Traverse all components in `tree` and accumulate all components where
+ * `test(component)` is `true`. This is not that useful on its own, but it's used
+ * as a primitive for other test utils.
+ *
+ * @export
+ * @param {Component<any, any>} root
+ * @param {(i: ReactInstance) => boolean} fn
+ * @returns {ReactInstance[]}
+ */
+export function findAllInRenderedTree(
+    root: Component<any, any>,
+    fn: (i: ReactInstance) => boolean): ReactInstance[];
+
+/**
+ * Finds all DOM elements of components in the rendered tree that are
+ * DOM components with the class name matching `className`.
+ *
+ * @export
+ * @param {Component<any, any>} root
+ * @param {string} className
+ * @returns {Element[]}
+ */
+export function scryRenderedDOMComponentsWithClass(
+    root: Component<any, any>,
+    className: string): Element[];
+/**
+ * Like `scryRenderedDOMComponentsWithClass()` but expects there to be one result,
+ * and returns that one result, or throws exception if there is any other
+ * number of matches besides one.
+ *
+ * @export
+ * @param {Component<any, any>} root
+ * @param {string} className
+ * @returns {Element}
+ */
+export function findRenderedDOMComponentWithClass(
+    root: Component<any, any>,
+    className: string): Element;
+
+/**
+ * Finds all DOM elements of components in the rendered tree that are
+ * DOM components with the tag name matching `tagName`.
+ *
+ * @export
+ * @param {Component<any, any>} root
+ * @param {string} tagName
+ * @returns {Element[]}
+ */
+export function scryRenderedDOMComponentsWithTag(
+    root: Component<any, any>,
+    tagName: string): Element[];
+/**
+ * Like `scryRenderedDOMComponentsWithTag()` but expects there to be one result,
+ * and returns that one result, or throws exception if there is any other
+ * number of matches besides one.
+ *
+ * @export
+ * @param {Component<any, any>} root
+ * @param {string} tagName
+ * @returns {Element}
+ */
+export function findRenderedDOMComponentWithTag(
+    root: Component<any, any>,
+    tagName: string): Element;
+
+/**
+ * Finds all instances of components with type equal to `componentClass`.
+ *
+ * @export
+ * @template T
+ * @template C
+ * @param {Component<any, any>} root
+ * @param {ClassType<any, T, C>} type
+ * @returns {T[]}
+ */
+export function scryRenderedComponentsWithType<T extends Component<{}, {}>, C extends ComponentClass<{}>>(
+    root: Component<any, any>,
+    type: ClassType<any, T, C>): T[];
+
+/**
+ * Same as `scryRenderedComponentsWithType()` but expects there to be one result
+ * and returns that one result, or throws exception if there is any other
+ * number of matches besides one.
+ *
+ * @export
```

Comment:
This (`@export`, `@template`, `@param` with no description, `@returns` with no description) just duplicates what's in the definition already.